File: 349.py
import random
nums = [3,2,1,4]
# no_extremes = [i for i in nums if i not in (min(nums),max(nums))]
# if bool(no_extremes) == False : answer = -1
# else : answer = random.choice(no_extremes)
# print(answer)

#lol, better:
if len(nums) < 3:
    print -1
nums.remove(min(nums))
nums.remove(max(nums))
print(random.choice(nums))

File: 349.js
const nums = [3,2,1,4];

if (nums.length < 3) {
    console.log(-1);
}
let iMax = nums.indexOf(Math.max(...nums))
let iMin = nums.indexOf(Math.min(...nums))
//splice() edits the orig array
nums.splice(iMax, 1);
nums.splice(iMin, 1);
//random item from an array
answer = nums[Math.floor(Math.random()*nums.length)]
console.log(answer)